### PR TITLE
Changed yaml reading to accomodate for possible errors

### LINKED ISSUE #167: WORKSPACE files keep going after error in parsing
In WORKSPACE files if error occurs during parsing of one of elements, bazel steward stops parsing rest of the elements. In such a case app should keep going

### PR DESCRIPTION
**Pull request by mikkoziel**
Changed yaml reading to accommodate for possible errors to ensure skipping it. Fix #167

### CODE REVIEW

File: kinds/bazel/rules/src/main/kotlin/org/virtuslab/bazelsteward/bazel/rules/BazelRulesExtractor.kt [hunk 1]
```diff
@@ -65,11 +65,29 @@ class BazelRulesExtractor(private val workspaceRoot: Path) {
...
       }
-      val result = yamlReader.readValue(resultFilePath.toFile(), object : TypeReference<List<Repository>>() {})
+
+      val yamlNode = yamlReader.readTree(resultFilePath.toFile())
+      val repositories = yamlNode.elements().asSequence().toList()
+        .filter {
+          it.has("kind") &&
+            it.get("kind").textValue() == "http_archive" &&
+            it.has("generator_function") &&
+            it.get("generator_function").textValue().isEmpty()
```
Comment (lukaszwawrzyk) [inline kinds/bazel/rules/src/main/kotlin/org/virtuslab/bazelsteward/bazel/rules/BazelRulesExtractor.kt:75]: It is easier to write and maintain, to first parse and then, using mapped objects do the filtering

File: kinds/bazel/rules/src/main/kotlin/org/virtuslab/bazelsteward/bazel/rules/BazelRulesExtractor.kt [hunk 2]
```diff
@@ -65,11 +65,29 @@ class BazelRulesExtractor(private val workspaceRoot: Path) {
...
+        }
+        .mapNotNull {
+          it.let {
```
Comment (lukaszwawrzyk) [inline kinds/bazel/rules/src/main/kotlin/org/virtuslab/bazelsteward/bazel/rules/BazelRulesExtractor.kt:78]: what is the purpose of it.let?

**Review by lukaszwawrzyk — CHANGES_REQUESTED** [review]

File: kinds/bazel/rules/src/main/kotlin/org/virtuslab/bazelsteward/bazel/rules/BazelRulesExtractor.kt (hunk 1 above)
Comment (mikkoziel) [reply kinds/bazel/rules/src/main/kotlin/org/virtuslab/bazelsteward/bazel/rules/BazelRulesExtractor.kt:75]: Fixed

File: kinds/bazel/rules/src/main/kotlin/org/virtuslab/bazelsteward/bazel/rules/BazelRulesExtractor.kt (hunk 2 above)
Comment (mikkoziel) [reply kinds/bazel/rules/src/main/kotlin/org/virtuslab/bazelsteward/bazel/rules/BazelRulesExtractor.kt]: Removed

**Pull request merged**